src/components/MatchRuleList.tsx

```diff
@@ -0,0 +1,255 @@
+
+      {isLoading ? (
+        <LoadingSpinner />
+      ) : matchRules.filter(r => !r.toDelete).length ? (
+        <ReactSortable
+          list={matchRules}
+          setList={setMatchRules}
+          className="card w-full divide-y divide-dashed divide-gray-600 overflow-visible text-left sm:space-y-1 sm:divide-none"
+          handle=".handler"
+        >
+          {matchRules.map(matchRule => (
+            <div
+              className={`group flex w-full flex-wrap items-center gap-x-1 gap-y-4 py-4 sm:gap-x-3 sm:gap-y-2 sm:py-2 ${
+                (matchRule.toDelete && 'hidden') || ''
+              }`}
+              key={matchRule.id}
+            >
+              <div>
+                <Bars2Icon className="icon-sm handler cursor-grab" />
+              </div>
+              <div className="flex flex-grow flex-col items-center space-y-1 sm:flex-row sm:gap-x-2">
+                <div className="flex-grow">
+                  <FormField
+                    inputWrapperClass="mt-0"
+                    name="match"
+                    label={t('matchRules.match')}
+                    labelClass="sr-only"
+                    type="text"
+                    value={matchRule.match ?? ''}
+                    onChange={match => {
+                      let newMatchRules = [...matchRules]
+                      const ruleIndex = matchRules.findIndex(
+                        rule => rule.id === matchRule.id
+                      )
+                      newMatchRules[ruleIndex].match = match.target.value
+                      setMatchRules(newMatchRules)
+                    }}
+                  />
+                </div>
+                <div className="hidden sm:block">
+                  <ArrowLongRightIcon className="icon-sm" />
+                </div>
+                <div className="sm:hidden">
+                  <ArrowLongDownIcon className="icon-sm" />
+                </div>
+                <div className="flex-grow">
+                  <Autocomplete<Account>
+                    inputWrapperClass="relative mt-0"
+                    groups={accountGroups}
+                    allowNewCreation={false}
+                    itemLabel={account => safeName(account, 'account')}
+                    itemId={account => account.id}
+                    label={t('accounts.account')}
+                    labelClass="sr-only"
+                    onChange={account => {
+                      let newMatchRules = [...matchRules]
+                      const ruleIndex = matchRules.findIndex(
+                        rule => rule.id === matchRule.id
+                      )
+                      newMatchRules[ruleIndex].accountId = account.id
+                      setMatchRules(newMatchRules)
+                    }}
+                    value={
+                      (accounts.find(
+                        account =>
+                          account.id === matchRule.accountId &&
+                          !account.archived
```

Comment:
Why are we explicitly excluding archived accounts here? What happens if the account is archived at some point between creating the match rule and then editing it later? If anything, the filtering should happen before we assign the list to `accounts`, not here.

Reply:
Yeah, good point. Removed the filtering with 8a9f43c, so now accounts are also correctly displayed when they are archived.